File: adapters/ruby/example/hash_connector.rb
# Reference urirun connector in Ruby: prints a urirun.bindings.v2 document.
require_relative "../urirun"

c = Urirun::Connector.new("hash", "hash")
c.command("sha256/command/file",
          { required: ["path"], properties: { "path" => { "type" => "string" } } },
          ["sha256sum", "{path}"])
puts c.bindings_json

File: adapters/perl/example/hash_connector.pl
# Reference urirun connector in Perl: prints a urirun.bindings.v2 document.
use strict;
use warnings;
use FindBin;
use lib "$FindBin::Bin/..";
use Urirun;

my $c = Urirun->new("hash", "hash");
$c->command(
    "sha256/command/file",
    { required => ["path"], properties => { path => { type => "string" } } },
    ["sha256sum", "{path}"],
);
print $c->bindings_json;
